File: 프로그래머스/Level3/네트워크.cpp
#include <string>
#include <vector>

using namespace std;

bool visit[200];

void dfs(vector<vector<int>> &computers, int n, int start){
    visit[start] = true;
    
    for(int i=0; i<n; i++){
        if(!visit[i] && computers[start][i] == 1){
            dfs(computers, n, i);
        }
    }
}

int solution(int n, vector<vector<int>> computers) {
    int answer = 0;
    
    for(int i=0; i<n; i++){
        if(!visit[i]){
            dfs(computers, n, i);
            answer ++;
        }
    }
    
    return answer;
}

File: 프로그래머스/Level3/네트워크.js
function solution(n, computers) {
    var answer = 0;
    let tmp = Array.from({length: 200}, () => false);
   
    function dfs(start){
        tmp[start] = true;
        
        for(let i=0; i<n; i++){
            if(!tmp[i] && computers[start][i] === 1){
                dfs(i);
            }    
        }
    }
    
    for(let j=0; j<n; j++){
        if(!tmp[j]){
            dfs(j);
            answer ++;
        }
    }
    
    return answer;
}
